packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -669,6 +671,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         runtimeOptions: IContainerRuntimeOptions = {},
         containerScope: FluidObject = context.scope,
         existing?: boolean,
+        containerRuntimeCtor: typeof ContainerRuntime = ContainerRuntime
```

Comment:
rather than adding an optional param to this method, could we encapsulate the functionality here in another function that takes a ctor, that is then called within this function. its a bit more roundabout, but something bothers me about a static function on a class taking a constructor for itself.

Reply:
i like extracting into a function as well as it keeps this simple route simple, no additional concepts, even if optional, but let's an advanced user do the advanced thing

Reply:
extracting it out as you suggested would require exposing the constructor further (this change goes private -> protected, that would make it public). I did consider making the variant of the method that takes in a constructor a static protected method on the class. That should have roughly the same bundle impact as your suggestion but keep the current signature for existing users. Thoughts?

Reply:
hmmm... yeah the initialization pattern used here is a bit hard to work with. I guess i'm fine with this change as is for now. need to think about how we could make the flow better here.